cli/azd/ci-build.ps1

```diff
@@ -1,5 +1,57 @@
 param(
-    [string] $Version,
-    [string] $SourceVersion
+    [string] $Version = (Get-Content "$PSScriptRoot/../version.txt"),
+    [string] $SourceVersion = (git rev-parse HEAD)
 )
-go build -ldflags="-X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)'"
\ No newline at end of file
+
+# On Windows, use the goversioninfo tool to embed the version information into the executable.
+if ($IsWindows) {
+    Write-Host "Windows build, set verison info and run 'go generate'"
+    if (! (Get-Command "goversioninfo" -ErrorAction SilentlyContinue)) {
+        Write-Host "goversioninfo not found, installing"
+        go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+        Get-Command "goversioninfo" -ErrorAction Stop
+    }
+
+    $VERSION_INFO_PATH = "$PSScriptRoot/versioninfo.json"
+
+    $exeFileVersion = ."$PSScriptRoot/../../eng/scripts/Get-MsiVersion.ps1" -CliVersion $Version
+    $splitExeFileVersion = $exeFileVersion -split '\.'
+    $versionInfo = Get-Content $VERSION_INFO_PATH | ConvertFrom-Json
+
+    $versionInfo.FixedFileInfo.FileVersion.Major = [int]$splitExeFileVersion[0]
+    $versionInfo.FixedFileInfo.FileVersion.Minor = [int]$splitExeFileVersion[1]
+    $versionInfo.FixedFileInfo.FileVersion.Patch = [int]$splitExeFileVersion[2]
+    # Build number is ignored in MSI upgrade logic
+    $versionInfo.FixedFileInfo.FileVersion.Build = 0
```

Comment:
That's only true of the `ProductVersion` property. All 4 quad parts matter for every other version within. I'd use the same algorithm - perhaps refactor it out into shared code - as we came up with for the MSI but in this case you can not bother multiplying by 100. Could, though, but you do have all 4 numbers to play with now.

Reply:
I'm using the same logic from `Get-MSIVersion.ps1` on line 17. We'll keep that same logic for now. 

But the comment is definitely inaccurate so I'll remove that.